### PR TITLE
POC Planif - Process monitoring screen - experimental feature (#4697)

### PR DESCRIPTION
**Pull request by vlo-rte**

### CODE REVIEW

File: ui/main/src/app/modules/monitoringexp/components/cell-renderers/sender-cell-renderer.component.html [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<!-- Copyright (c) 2022, RTE (http://www.rte-france.com)                   -->
```
[the remark this answers is not in the document]
Comment (vlo-rte) [reply ui/main/src/app/modules/monitoringexp/components/cell-renderers/sender-cell-renderer.component.html:1]: Done.

File: ui/main/src/app/modules/monitoringexp/components/cell-renderers/sender-cell-renderer.component.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
+/* Copyright (c) 2022, RTE (http://www.rte-france.com)
```
[the remark this answers is not in the document]
Comment (vlo-rte) [reply ui/main/src/app/modules/monitoringexp/components/cell-renderers/sender-cell-renderer.component.ts:1]: Done.

File: ui/main/src/app/modules/monitoringexp/components/cell-renderers/state-cell-renderer.component.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    agInit(params: any): void {
+        this.stateColor = {'color': params.data.stateColor};
+        this.stateLabel = params.context.componentParent.processStateNameMap.get(params.data.process + '.' + params.data.state);
+    }
+
+    // noinspection JSUnusedLocalSymbols
+    /** This method returns true to signal to the grid that this renderer doesn't need to be recreated if the underlying data changes
+     *  See https://www.ag-grid.com/documentation/angular/component-cell-renderer/#handling-refresh
+     * */
+    refresh(params: ICellRendererParams): boolean {
```
[the remark this answers is not in the document]
Comment (vlo-rte) [reply ui/main/src/app/modules/monitoringexp/components/cell-renderers/state-cell-renderer.component.ts:35]: Done.

File: ui/main/src/app/modules/monitoringexp/components/cell-renderers/state-description-cell-renderer.component.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    // noinspection JSUnusedLocalSymbols
+    /** This method returns true to signal to the grid that this renderer doesn't need to be recreated if the underlying data changes
+     *  See https://www.ag-grid.com/documentation/angular/component-cell-renderer/#handling-refresh
+     * */
+    refresh(params: ICellRendererParams): boolean {
+        return true;
+    }
+}
```
[the remark this answers is not in the document]
Comment (vlo-rte) [reply ui/main/src/app/modules/monitoringexp/components/cell-renderers/state-description-cell-renderer.component.ts:41]: Done.

File: ui/main/src/app/modules/monitoringexp/components/cell-renderers/state-description-cell-renderer.component.html [hunk 5]
```diff
@@ -0,0 +1,9 @@
+<!-- Copyright (c) 2022, RTE (http://www.rte-france.com)                   -->
+<!-- See AUTHORS.txt                                                       -->
+<!-- This Source Code Form is subject to the terms of the Mozilla Public   -->
+<!-- License, v. 2.0. If a copy of the MPL was not distributed with this   -->
+<!-- file, You can obtain one at http://mozilla.org/MPL/2.0/.              -->
+<!-- SPDX-License-Identifier: MPL-2.0                                      -->
+<!-- This file is part of the OperatorFabric project.                      -->
+
```
[the remark this answers is not in the document]
Comment (vlo-rte) [reply ui/main/src/app/modules/monitoringexp/components/cell-renderers/state-description-cell-renderer.component.html:8]: Done.

File: ui/main/src/app/modules/monitoringexp/monitoringexp.component.ts [hunk 6]
```diff
@@ -0,0 +1,300 @@
...
+                value: process.id,
+                label: itemName,
+                i18nPrefix: `${process.id}.${process.version}`
+            });
+        });
+    }
+
+    ngOnInit() {
+        this.size = this.configService.getConfigValue('logging.filters.page.size', 10);
+        this.tags = this.configService.getConfigValue('logging.filters.tags.list');
```
[the remark this answers is not in the document]
Comment (vlo-rte) [reply ui/main/src/app/modules/monitoringexp/monitoringexp.component.ts]: already done if I'm not wrong.

File: ui/main/src/assets/i18n/fr.json [hunk 7]
```diff
@@ -100,6 +100,7 @@
     "logging": "Logging",
     "admin": "Administration",
     "monitoring": "Monitoring",
+    "monitoringexp": "Monitoring exp.",
```
[the remark this answers is not in the document]
Comment (vlo-rte) [reply ui/main/src/assets/i18n/fr.json:103]: Done.

File: ui/main/src/assets/i18n/en.json [hunk 8]
```diff
@@ -100,6 +100,7 @@
     "logging": "Logging",
     "admin": "Administration",
     "monitoring": "Monitoring",
+    "monitoringexp": "Monitoring exp.",
```
[the remark this answers is not in the document]
Comment (vlo-rte) [reply ui/main/src/assets/i18n/en.json:103]: Done.

File: ui/main/src/app/modules/monitoringexp/monitoringexp.component.ts [hunk 9]
```diff
@@ -0,0 +1,315 @@
...
+        this.loadingInProgress = true;
+
+        const {value} = this.monitoringExpForm;
+        this.filtersTemplate.transformFiltersListToMap(value);
+
+        const filter = this.getFilter(page_number, this.size, this.filtersTemplate.filters);
+
+        this.cardService
+            .fetchFilteredCards(filter)
+            .pipe(takeUntil(this.unsubscribe$))
```
Comment (freddidierRTE) [inline ui/main/src/app/modules/monitoringexp/monitoringexp.component.ts:162]: You do not need to unsubscribe as the observable is close automatically in this case

File: ui/main/src/app/modules/monitoringexp/monitoringexp.component.ts [hunk 10]
```diff
@@ -0,0 +1,315 @@
...
+            backdrop: 'static', // Modal shouldn't close even if we click outside it
+            size: 'sm'
+        };
+        this.modalRef = this.modalService.open(this.exportTemplate, modalOptions);
+
+        const filter = this.getFilter(0, this.resultsNumber, this.filtersTemplate.filters);
+
+        this.cardService
+            .fetchFilteredCards(filter)
+            .pipe(takeUntil(this.unsubscribe$))
```
Comment (freddidierRTE) [inline ui/main/src/app/modules/monitoringexp/monitoringexp.component.ts:272]: You do not need to unsubscribe as the observable is close automatically in this case

**Review by freddidierRTE — CHANGES_REQUESTED** [review]

**Comment by freddidierRTE** [comment]
replace monitoringexp by processMonitoring
rebase code 
squash commits 
set back version to SNAPSHOT

**Comment by vlo-rte** [comment]
> replace monitoringexp by processMonitoring rebase code squash commits set back version to SNAPSHOT

Done.

File: config/docker/ui-config/ui-menu.json [hunk 11]
```diff
@@ -12,6 +12,9 @@
     {
       "opfabCoreMenuId": "monitoring"
     },
+    {
+      "opfabCoreMenuId": "monitoringexp"
```
[the remark this answers is not in the document]
Comment (vlo-rte) [reply config/docker/ui-config/ui-menu.json]: Done.

File: ui/main/src/app/modules/monitoringexp/components/monitoringexp-table/monitoringexp-table.component.ts [hunk 12]
```diff
@@ -0,0 +1,244 @@
...
+        this.translate
+            .get(key, interpolateParams)
+            .pipe(takeUntil(this.unsubscribe$))
+            .subscribe((translate) => {
+                translatedColumn = translate;
+            });
+
+        return translatedColumn;
+    }
+
```
[the remark this answers is not in the document]
Comment (vlo-rte) [reply ui/main/src/app/modules/monitoringexp/components/monitoringexp-table/monitoringexp-table.component.ts]: Done. Removed.

File: ui/main/src/app/modules/monitoringexp/monitoringexp.component.ts (hunk 9 above)
Comment (vlo-rte) [reply ui/main/src/app/modules/monitoringexp/monitoringexp.component.ts]: Done. Removed.

File: ui/main/src/app/modules/monitoringexp/monitoringexp.component.ts (hunk 10 above)
Comment (vlo-rte) [reply ui/main/src/app/modules/monitoringexp/monitoringexp.component.ts]: Done. Removed.

File: ui/main/src/app/modules/monitoringexp/monitoringexp.component.ts [hunk 13]
```diff
@@ -0,0 +1,315 @@
...
+
+                    exportArchiveData.push(lineForExport);
+                });
+                ExcelExport.exportJsonToExcelFile(exportArchiveData, 'MonitoringExp');
+                this.modalRef.close();
+            });
+    }
+
+    translateColumn(key: string, interpolateParams?: Map<string,string>): any {
+        if (!key) return '';
```
[the remark this answers is not in the document]
Comment (vlo-rte) [reply ui/main/src/app/modules/monitoringexp/monitoringexp.component.ts]: Done.

File: ui/main/src/app/modules/monitoringexp/components/cell-renderers/sender-cell-renderer.component.ts [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+ */
+
+import {Component} from '@angular/core';
+import {ICellRendererAngularComp} from 'ag-grid-angular';
+import {ICellRendererParams} from 'ag-grid-community';
+
+@Component({
+    selector: 'of-sender-cell-renderer',
+    templateUrl: './sender-cell-renderer.component.html'
+})
```
[the remark this answers is not in the document]
Comment (vlo-rte) [reply ui/main/src/app/modules/monitoringexp/components/cell-renderers/sender-cell-renderer.component.ts]: Done.

File: ui/main/src/app/modules/monitoringexp/components/cell-renderers/time-cell-renderer.component.ts [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+ */
+
+import {Component} from '@angular/core';
+import {DateTimeFormatterService} from 'app/business/services/date-time-formatter.service';
+import {ICellRendererAngularComp} from 'ag-grid-angular';
+import {ICellRendererParams} from 'ag-grid-community';
+
+@Component({
+    selector: 'of-time-cell-renderer',
+    templateUrl: './time-cell-renderer.component.html'
```
[the remark this answers is not in the document]
Comment (vlo-rte) [reply ui/main/src/app/modules/monitoringexp/components/cell-renderers/time-cell-renderer.component.ts]: Done.

File: ui/main/src/app/modules/monitoringexp/components/monitoringexp-table/monitoringexp-table.component.ts [hunk 16]
```diff
@@ -0,0 +1,244 @@
...
+import {LightCard} from '@ofModel/light-card.model';
+import {TimeCellRendererComponent} from '../cell-renderers/time-cell-renderer.component';
+import {SenderCellRendererComponent} from '../cell-renderers/sender-cell-renderer.component';
+import {Subject} from 'rxjs';
+import {NgbModal, NgbModalOptions, NgbModalRef} from "@ng-bootstrap/ng-bootstrap";
+import {SelectedCardService} from "../../../../business/services/card/selectedCard.service";
+
+@Component({
+    selector: 'of-monitoringexp-table',
+    templateUrl: './monitoringexp-table.component.html'
```
[the remark this answers is not in the document]
Comment (vlo-rte) [reply ui/main/src/app/modules/monitoringexp/components/monitoringexp-table/monitoringexp-table.component.ts]: Done.

File: ui/main/src/app/modules/monitoringexp/monitoringexp.component.ts [hunk 17]
```diff
@@ -0,0 +1,315 @@
...
+import {CardsFilter} from '@ofModel/cards-filter.model';
+import {FilterMatchTypeEnum, FilterModel} from '@ofModel/filter-model';
+import {CardService} from 'app/business/services/card/card.service';
+import {TranslationService} from 'app/business/services/translation/translation.service';
+import {SelectedCardService} from "../../business/services/card/selectedCard.service";
+
+@Component({
+    selector: 'of-monitoringexp',
+    templateUrl: './monitoringexp.component.html',
+    styleUrls: ['./monitoringexp.component.scss']
```
[the remark this answers is not in the document]
Comment (vlo-rte) [reply ui/main/src/app/modules/monitoringexp/monitoringexp.component.ts]: Done.

**Comment by freddidierRTE** [comment]
Issue with table rendering : 

image

**Review by freddidierRTE — CHANGES_REQUESTED** [review]

File: ui/main/src/app/modules/processmonitoring/processmonitoring.component.ts [hunk 18]
```diff
@@ -0,0 +1,318 @@
...
+                    });
+                    this.results = page.content;
+                    this.totalElements= page.totalElements;
+                    this.totalPages= page.totalPages;
+                    this.changeDetector.markForCheck();
+                },
+                error: () => {
+                    this.firstQueryHasBeenDone = false;
+                    this.loadingInProgress = false;
+                    this.technicalError = true;
```
[the remark this answers is not in the document]
Comment (vlo-rte) [reply ui/main/src/app/modules/processmonitoring/processmonitoring.component.ts:191]: Done.

**Pull request merged**